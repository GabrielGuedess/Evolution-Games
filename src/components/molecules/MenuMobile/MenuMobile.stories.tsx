import * as Popover from '@radix-ui/react-popover';
import { Story, Meta } from '@storybook/react/types-6-0';
import { CaretDown } from 'phosphor-react';

import { MenuMobile, MenuProps } from './MenuMobile';

export default {
  title: 'Molecules/MenuMobile',
  component: MenuMobile,
  parameters: {
    viewport: {
      defaultViewport: 'mobile1',
    },
  },
  args: {
    path: '/',
  },
  argTypes: {
    path: {
      control: {
        type: 'select',
        options: ['NotActive', '/', '/explorer', '/contact'],
      },
    },
    username: {
      control: {
        type: 'select',
        options: ['User', null],
      },
    },
  },
} as Meta<MenuProps>;

export const Default: Story<MenuProps> = args => (
  <Popover.Root>
    <Popover.Trigger>
      <CaretDown aria-label="Icon" size={32} style={{ cursor: 'pointer' }} />
    </Popover.Trigger>
    <MenuMobile {...args} />
  </Popover.Root>
);
