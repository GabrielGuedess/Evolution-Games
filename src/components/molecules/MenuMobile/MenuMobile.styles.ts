import * as NavigationMenu from '@radix-ui/react-navigation-menu';
import * as Popover from '@radix-ui/react-popover';
import media from 'styled-media-query';

import styled, { css } from 'styled-components';

export const Wrapper = styled(Popover.Content)`
  ${({ theme }) => css`
    width: calc(100vw - (1.6rem * 2));
    display: flex;
    flex-direction: column;
    justify-content: center;
    padding: 4rem 1.2rem;
    margin: 0 1.6rem;
    background: rgba(4, 4, 12, 0.8);
    backdrop-filter: blur(0.3rem);
    border-radius: ${theme.border.radius.regular};

    ${media.greaterThan('medium')`
      display: none;
    `}
  `}
`;

export const ListTextLink = styled(NavigationMenu.List)`
  display: flex;
  flex-direction: column;
  align-items: center;
  gap: 3.2rem;
`;

export const ItemTextLink = styled(NavigationMenu.Item)`
  list-style: none;
`;

export const LinkTextLink = styled(NavigationMenu.Link)`
  ${({ theme, active }) => css`
    text-decoration: none;
    outline: none;
    font-weight: 500;
    font-size: ${theme.font.sizes.small};
    line-height: 3.2rem;
    text-transform: uppercase;
    transition: color 0.3s linear;

    color: ${active ? theme.colors.primary : theme.colors.whiteText};
    text-shadow: ${active && theme.shadows.text};

    :is(:hover, :focus) {
      color: ${theme.colors.primary};
      text-shadow: ${theme.shadows.text};
    }
  `}
`;
