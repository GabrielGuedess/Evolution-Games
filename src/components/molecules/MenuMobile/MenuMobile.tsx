import { useSession } from 'next-auth/react';
import Link from 'next/link';

import * as NavigationMenu from '@radix-ui/react-navigation-menu';

import Button from 'components/atoms/Button/Button';

import * as S from './MenuMobile.styles';

export type MenuProps = {
  path: string;
};

export const MenuMobile = ({ path }: MenuProps) => {
  const { data: session } = useSession();

  return (
    <S.Wrapper>
      <NavigationMenu.Root>
        <S.ListTextLink>
          <S.ItemTextLink>
            <Link href="/" passHref>
              <S.LinkTextLink active={path === '/'}>Home</S.LinkTextLink>
            </Link>
          </S.ItemTextLink>

          <S.ItemTextLink>
            <Link href="/explorer" passHref>
              <S.LinkTextLink active={path === '/explorer'}>
                Explorer
              </S.LinkTextLink>
            </Link>
          </S.ItemTextLink>

          <S.ItemTextLink>
            <Link href="/" passHref>
              <S.LinkTextLink active={path === '/contact'}>
                Contact
              </S.LinkTextLink>
            </Link>
          </S.ItemTextLink>

          <S.ItemTextLink>
            <Link
              href={!!session?.user.username ? '/profile/me' : '/sign-in'}
              passHref
            >
              <Button size="large" as="a">
                {!!session?.user.username ? 'View Profile' : 'Sign In'}
              </Button>
            </Link>
          </S.ItemTextLink>
        </S.ListTextLink>

        <NavigationMenu.Viewport />
      </NavigationMenu.Root>
    </S.Wrapper>
  );
};
