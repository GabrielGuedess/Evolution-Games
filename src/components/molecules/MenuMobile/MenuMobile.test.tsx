import * as Popover from '@radix-ui/react-popover';
import { renderWithTheme } from 'utils/tests/helpers';

import { fireEvent, screen } from '@testing-library/react';

import { MenuMobile } from './MenuMobile';

global.ResizeObserver = jest.fn().mockImplementation(() => ({
  observe: jest.fn(),
  unobserve: jest.fn(),
  disconnect: jest.fn(),
}));

describe('<MenuMobile />', () => {
  it('should render user logged', () => {
    // Arrange
    const { container } = renderWithTheme(
      <Popover.Root>
        <Popover.Trigger>
          <div data-testid="Click" />
        </Popover.Trigger>
        <MenuMobile path="/" username="Gabriel" />
      </Popover.Root>,
    );

    // Act
    fireEvent.click(screen.getByTestId('Click'));

    // Assert
    expect(screen.getByText('View Profile')).toBeInTheDocument();
    expect(screen.queryByText('Sign In')).not.toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render user not logged', () => {
    renderWithTheme(
      <Popover.Root>
        <Popover.Trigger>
          <div data-testid="Click" />
        </Popover.Trigger>
        <MenuMobile path="/" />
      </Popover.Root>,
    );

    // Act
    fireEvent.click(screen.getByTestId('Click'));

    // Assert
    expect(screen.getByText('Sign In')).toBeInTheDocument();
    expect(screen.queryByText('View Profile')).not.toBeInTheDocument();
  });
});
