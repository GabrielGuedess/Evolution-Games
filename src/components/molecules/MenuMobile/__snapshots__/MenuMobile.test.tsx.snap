// Jest Snapshot v1, https://goo.gl/fbAQLP

exports['<MenuMobile /> should render user logged 1'] = `
@media (min-width:450px) {

}

@media (min-width:768px) {

}

<button
  aria-controls="radix-:r0:"
  aria-expanded="true"
  aria-haspopup="dialog"
  data-state="open"
  type="button"
>
  <div
    data-testid="Click"
  />
</button>
`;
